docker/debian11/compile/Dockerfile

```diff
@@ -0,0 +1,120 @@
+FROM debian:11
+ENV TZ=Asia/Shanghai
+ENV GITHUB_PROXY=https://ghproxy.com/
+ENV VERSION_ETCD=v3.4.0 \
+    DOWNLOAD_ETCD_URL=${GITHUB_PROXY}https://github.com/etcd-io/etcd/releases/download \
+    GOPROXY=https://goproxy.io,direct \
+    PROTOC_VERSION=21.8 \
+    GO_INSTALL_VERSION=1.19.6 \
+    PROTOC_GEN_GO_VERSION=v1.28 \
+    PROTOC_GEN_GO_GRPC_VERSION=v1.2
+RUN sed -i 's/deb.debian.org/mirrors.ustc.edu.cn/g' /etc/apt/sources.list && \
+    sed -i 's|security.debian.org/debian-security|mirrors.ustc.edu.cn/debian-security|g' /etc/apt/sources.list && \
+    apt clean && \
+    apt -y update && \
+    apt -y install \
+        g++ \
+        wget \
+        clang \
+        git \
+        libnl-genl-3-dev \
+        libssl-dev \
+        uuid-dev \
+        libcurl4-gnutls-dev \
+        libfiu-dev \
+        openjdk-11-jdk \
+        sudo \
+        python3-dev \
+        python3-setuptools \
+        python3-wheel \
+        zlib1g-dev \
+        libsnappy-dev \
+        make \
+        liblz4-dev \
+        ca-certificates \
+        curl \
+        gnupg \
+        lsb-release \
+        curl \
+        apt-transport-https \
+        git \
+        libbison-dev \
+        flex \
+        libhashkit-dev \
+        libevent-dev \
+        procps \
+        lsof \
+        gdb \
+        unzip \
+        musl-tools \
+        libunwind8-dev \
+        cmake \
+        fuse3 \
+        libfuse3-dev \
+        libnl-3-200\ 
+        libnl-3-dev \
+        libnl-genl-3-200 \
+        libnl-genl-3-dev  \
+        libjemalloc2 \
+        file \
+        lcov \
+        bc \
+        gcovr && \
+    install -m 0755 -d /etc/apt/keyrings && \
+    curl -fsSL https://download.docker.com/linux/debian/gpg | \
+    gpg --dearmor -o /etc/apt/keyrings/docker.gpg && \
+    chmod a+r /etc/apt/keyrings/docker.gpg && \
+    echo "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/debian \
+    "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" | \
+    tee /etc/apt/sources.list.d/docker.list > /dev/null && \
+    apt update && \
+    apt install -y \
+        docker-ce \
+        docker-ce-cli \
+        containerd.io \
+        docker-buildx-plugin \
+        docker-compose-plugin
+RUN wget ${GITHUB_PROXY}https://github.com/bazelbuild/bazelisk/releases/download/v1.11.0/bazelisk-linux-amd64 && \
+    mv bazelisk-linux-amd64 /usr/bin/bazelisk && \
+    ln -sf /usr/bin/bazelisk /usr/bin/bazel && \
+    wget ${GITHUB_PROXY}https://github.com/bazelbuild/bazel/releases/download/4.2.2/bazel-4.2.2-linux-x86_64 && \
+    mkdir -p /root/.cache/bazelisk/downloads/bazelbuild/bazel-4.2.2-linux-x86_64/bin && \
```

Comment:
you can set another environment, `BAZELISK_BASE_URL=https://repo.huaweicloud.com/bazel/`, so that bazelisk will download corresponding bazel from this url

Reply:
done

> you can set another environment, `BAZELISK_BASE_URL=https://repo.huaweicloud.com/bazel/`, so that bazelisk will download corresponding bazel from this url